src/tracing/mux.rs

```diff
@@ -0,0 +1,388 @@
+use crate::tracing::{FourByteInspector, TracingInspector, TracingInspectorConfig};
+use alloy_primitives::{Address, Log, U256};
+use alloy_rpc_trace_types::geth::{
+    mux::{MuxConfig, MuxFrame},
+    CallConfig, FourByteFrame, GethDebugBuiltInTracerType, GethDebugTracerConfig, GethTrace,
+    NoopFrame, PreStateConfig,
+};
+use revm::{
+    interpreter::{CallInputs, CallOutcome, CreateInputs, CreateOutcome, Interpreter},
+    primitives::ResultAndState,
+    Database, DatabaseRef, EvmContext, Inspector,
+};
+use std::collections::HashMap;
+use thiserror::Error;
+
+/// Mux tracing inspector that runs and collects results of multiple inspectors at once.
+#[derive(Clone, Debug)]
+pub struct MuxInspector {
+    /// The list of tracer types with its inspectors.
+    inspectors: Vec<(GethDebugBuiltInTracerType, MultiInspector)>,
+}
+
+impl MuxInspector {
+    /// Try creating a new instance of [MuxInspector] from the given [MuxConfig].
+    pub fn try_from_config(config: MuxConfig) -> Result<MuxInspector, Error> {
```

Comment:
hmm, maybe we should actually make MultiInspector pub so this type can be configured easily manually without needing MuxConfig with an additional new function

```
pub new(impl IntoIter<MultiInspector>) -> Self
```

but not required for this pr